### ISSUE TITLE
Qute fails to detect src/main/resources templates when running maven test

### ISSUE THREAD

**Issue by mjgkaastrupandersen** [quarkusio/quarkus#17271]
I have tried implementing my Qute templates using `@ResourcePath` annotation but I am switching to the Type-safe templates strategy to have compile checking (which works great btw).

Maven compile reports build success but when running a maven test, my tests fail, as my template-using classes no longer is able to find the templates:

```
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.qute.deployment.QuteProcessor#collectTemplateTypeInfo threw an exception: io.quarkus.qute.TemplateException: Declared template ClassName/templateName could not be found. Either add it or delete its declaration in [...]ClassName$Templates.templateName
```

I assume that this is because, in test, Qute is looking for the template in the src/**test**/resources/templates/ClassName/templateName and no longer src/**main**/resources/templates/ClassName/templateName as it did on compile-time.

Copying my templates to src/**test**/resources/templates, the issue no longer exists but this is not very desirable, as my tests no longer would test production templates.

This is not expected behaviour in my opinion. 

Also, I would like to use test-specific templates declared only in my test classes, which should probably be taken into consideration, if this issue results in a fix.

**Comment by mkouba** [comment]
Hello @mjgkaastrupandersen, a minimal reproducer and some information about your environment (e.g. Quarkus version) would be really helpful.

**Comment by michaelgarde** [comment]
@mkouba off course. My bad.
We are currently on 
* Quarkus version 1.11.6.Final-redhat-00001, 
* Maven 3.8.1, 
* java-11-openjdk 11.0.10
Build system is running on Ubuntu in WSL2

I'll see if I can find the time to create a minimal reproducer.

**Comment by famod** [comment]
I was just having the same problem, mainly while working in Eclipse.

I started with a test with a template in `src/test/resources/templates` and when it was green I continued to add a `Template` injection point to a class in `src/main/java` and I placed the respective template in `src/main/resources/templates`.
That's when trouble started. The respective test (a different one than I first mentioned) failed all the time with "No template found for ...". I even double check with `mvn`, same problem.
The test passed when I moved the template to `src/test/resources/templates` and it broke again when moving it back to main.

Then I ran `mvn clean ...` and I was finally able to run the test as expected!
So it seems there is a weird "caching" problem when you start with a template in test and then switch over to main.
